tests/test_serialization.py

```diff
@@ -217,3 +224,93 @@ def test_serialize_fail_llla_different_last_layer_name(laplace, model, model3, r
 
     with pytest.raises(ValueError):
         la2.load_state_dict(torch.load('state_dict.bin'))
+
+
+@pytest.mark.parametrize(
+    "model_device_str",
+    ["cpu", "cuda"] if torch.cuda.is_available() else ["cpu"],
+)
+@pytest.mark.parametrize(
+    "map_location",
+    ["cuda", "cpu"] if torch.cuda.is_available() else ["cpu"],
+)
+@pytest.mark.parametrize("laplace", flavors + flavors_subnet)
+def test_map_location(
+    laplace, model, reg_loader, tmp_path, map_location, model_device_str
+):
+    # Only for LLLaplace, we need dill.
+    #   AttributeError: Can't pickle local object 'FeatureExtractor._get_hook.<locals>.hook'
+    if issubclass(laplace, LLLaplace):
+        if find_spec("dill") is None:
+            pytest.skip(
+                reason="dill package not found but needed for this test"
+            )
+        else:
+            import dill
+
+            torch_save = lambda obj, fn: torch.save(
+                obj, fn, pickle_module=dill
+            )
+    else:
+        # Use default pickle_module=pickle, but no need to import pickle here
+        # just to set pickle_module=pickle.
+        torch_save = lambda obj, fn: torch.save(obj, fn)
```

Comment:
The usage of `dill` for LLLaplace to pass tests is optional and those will be skipped with a message. Is this OK? I'd not make `dill` a requirement, but maybe the developers would like to see a warning here in addition to the skip?

Reply:
The supported API so far is `torch`-compatible as in

```py
torch.save(obj, filename)
```

So, `obj`, being a `Laplace` instance, would need to

* figure out that it is being serialized
* in that case, catch any `AttributeError` such as

  ```
  AttributeError: Can't pickle local object 'FeatureExtractor._get_hook.<locals>.hook'
  ```

  Here one must make the assumption that any `AttributeError` is actually caused by an attempted `torch.save()` call.

I understand the reasoning behind your suggestion. However I don't know, off the top of my head, how to implement this. I'm happy for any hints here.

One solution that comes to mind would be to have a `laplace.save()` function that wraps `torch.save()` but extends the API, kind of breaks `torch` compatibility and needs to be documented.

The other option is to document potential serialization fails with plain `pickle` and advise users to use `torch.save(..., pickle_module=dill)`. In either case, users must be made aware via documentation.

Reply:
I'm leaning towards the documentation-based approach.

How about writing a new example specifically for serialization on README.md? Currently the Laplace-specific serialization example is coupled with another example:(https://github.com/aleximmer/Laplace/blob/2543274aad34e2612d708a58c55272fadbc41626/README.md?plain=1#L43-L76

Let's extract the serialization code from there, and add your `torch`-native-but-not-portable alternative. We can put this new example after  the `### Applying the LA over only a subset of the model parameters` example.

Reply:
I added a section to the README in 807644071078e8fb3bf41312c695dd1e762f4fa9 as you suggested. I feel that this may be a bit too technical for the README, should this instead go into the docs over at https://aleximmer.github.io/Laplace/?